File: plugin/integration/scala-plugin/src/scala/org/intellij/plugin/zeppelin/scala/cell/ZeppelinIdProvider.scala
package org.intellij.plugin.zeppelin.scala.cell

import com.intellij.psi.PsiElement
import org.intellij.plugin.zeppelin.scala.constants.ZeppelinScalaConstants
import org.jetbrains.plugins.scala.worksheet.cell.WorksheetCellExternalIdProvider

class ZeppelinIdProvider extends WorksheetCellExternalIdProvider {
  override def canHandle(startElement: PsiElement): Boolean = {
    startElement.getText.contains(ZeppelinScalaConstants.EXTERNAL_ID)
  }

  override def getId(startElement: PsiElement): Option[String] = {
    val text = startElement.getText
    val start = text.indexOf(ZeppelinScalaConstants.EXTERNAL_ID) + ZeppelinScalaConstants.EXTERNAL_ID.length
    val id = text.substring(start)
    Some(id)
  }
}

File: zeppelin/integration/scala-plugin/src/org/intellij/plugin/zeppelin/scala/cell/ZeppelinIdProvider.kt
package org.intellij.plugin.zeppelin.scala.cell

import com.intellij.psi.PsiElement
import org.intellij.plugin.zeppelin.scala.constants.ZeppelinScalaConstants
import org.jetbrains.plugins.scala.worksheet.cell.WorksheetCellExternalIdProvider
import scala.Option
import scala.Some

class ZeppelinIdProvider : WorksheetCellExternalIdProvider() {
  override fun canHandle(startElement: PsiElement): Boolean {
    return startElement.text.contains(ZeppelinScalaConstants.EXTERNAL_ID)
  }
  override fun getId(startElement: PsiElement): Option<String>? {
     val text: String = startElement.text
     val start: Int = text.indexOf(ZeppelinScalaConstants.EXTERNAL_ID) + ZeppelinScalaConstants.EXTERNAL_ID.length
     val id: String = text.substring(start)
    return Some(id)
  }
}